src/package-modules.js

```diff
@@ -376,6 +392,17 @@ async function createComposerJsonOnlyPackage(instruction, release, name, version
     isExecutable: false,
   }];
 
+  // Convert branch names to proper dev-branch format for Composer
+  let versionForFilename = release || ref;
```

Comment:
I don't think `release` would be correct here, after the refactor. Possibly `release.version` or `release.ref` or similar. Actually, `ref` itself isn't defined at all here that I can see. So both need to be updated/corrected. Same applies through line 399 below.